packages/next/src/server/lib/incremental-cache/file-system-cache.ts

```diff
@@ -110,24 +110,30 @@ export default class FileSystemCache implements CacheHandler {
     // we need to ensure the tagsManifest is refreshed
     // since separate workers can be updating it at the same
     // time and we can't flush out of sync data
-    this.loadTagsManifest()
-    if (!tagsManifest || !this.tagsManifestPath) {
+    if (!this.tagsManifestPath) {
       return
     }
+    const manifestWriteStream = this.fs.createWriteStream(this.tagsManifestPath)
```

Comment:
Why is this change needed?

Reply:
@ijjk it is as I described, we call multiple `revalidateTag` (`staticGenerationStore.pendingRevalidates`) with `await Promise.all`, but it is not safe to write to one file in a parallel manner.